core/sql/parser/sqlparser.y

```diff
@@ -7001,6 +7036,46 @@ rel_subquery_and_as_clause : rel_subquery as_clause
                                $$ = $2;
                              }
 
+with_clause_list : with_clause
+                     {  $$ = $1 ; }
+
+                   | with_clause_list ',' correlation_name TOK_AS rel_subquery 
```

Comment:
Can we do this in a way that doesn't have two different rules for the first and the subsequent parts of the WITH clause? See comment above for how the ANSI standard handles the WITH clause.